app/helpers/passport.js

```diff
@@ -19,36 +22,87 @@ module.exports = function(passport) {
     passport.use(new GitHubStrategy({
         clientID: config.authentication.github.clientId,
         clientSecret: config.authentication.github.clientSecret,
-        callbackURL: '/login/callback'
+        scope: 'repo, user'
     },
-    function(accessToken, refreshToken, profile, done) {
-        process.nextTick(function () {
-            User.findOne({ 'id':  profile.id }, function(err, user) {
-                if (err) {
-                    return done(err);
-                }
+    function(accessToken, refreshToken, githubProfile, done) {
+        User.findOne({ 'githubId': githubProfile.id }, function(err, user) {
+            if (err) {
+                return handleError(err);
+            }
+
+            if (user) {
+                return done(null, user);
+            } else {
+                var newAccount = new Account({
+                    name: 'hashtag-include'
+                });
 
-                if (user) {
-                    return done(null, user);
-                } else {
-                    var newUser = new User();
-
-                    newUser.id = profile.id;
-                    newUser.username = profile.login;
-                    newUser.email = profile.email;
-                    newUser.name = profile.name;
-                    newUser.avatar = profile.avatar_url; // jshint ignore:line
-
-                    // save the user
-                    newUser.save(function(err) {
-                        if (err) {
-                            throw err;
+                // save the user
+                newAccount.save(function(err) {
+                    if (err) {
+                        return handleError(err);
+                    }
+
+                    var newUser = new User({
+                        githubId: githubProfile.id,
+                        username: githubProfile._json.login,
+                        name: githubProfile._json.name,
+                        avatar: githubProfile._json.avatar_url, // jshint ignore:line
+                        account: newAccount._id
+                    });
+
+                    // retrieve the user's email address
+                    var requestOptions = {
+                        url: 'https://api.github.com/user/emails?access_token=' + accessToken,
+                        headers: {
+                            'User-Agent': 'hashtag-include'
```

Comment:
No, [it's required with v3 of the GitHub API](https://developer.github.com/v3/#user-agent-required). If you don't include it, you receive 403 response. It needs to be either the name of the application, or the name of the GitHub username.